src/boot/efi/boot.c

```diff
@@ -596,6 +607,10 @@ static EFI_STATUS reboot_into_firmware(void) {
         return log_error_status_stall(err, L"Error calling ResetSystem: %r", err);
 }
 
+static EFI_STATUS reboot_into_firmware(void) {
```

Comment:
this looks like the wrong way round. the reboot_into_firmware_entry() function should call into reboot_into_firmware() which implements the actually relevant code.